docgen/src/guide/Server-side_rendering.md

```diff
@@ -6,8 +6,24 @@ category: guide
 navWeight: 30
 ---
 
-React InstantSearch is not yet compatible with server-side rendering. This is one of our priority topics
-right now. We will soon provide API entries to handle this usecase.
+React InstantSearch builds the Algolia query based on every widget declared. That's why at the first rendering there are no results yet. This is inconvenient when doing server side rendering as the html generated by the server will not contain any hits. 
```

Comment:
I would host the app on zeit (or any fancy host), then link it from here "Live demo", "Code".

After this, some paragraph about how it's working and how to implement it without zeit (small example, high level example).

And then some details about why it's done like that (first render undefined, declarative API handles state etc..)